utils/crypto/secp256k1r.go

```diff
@@ -69,7 +69,7 @@ func (*FactorySECP256K1R) ToPublicKey(b []byte) (PublicKey, error) {
 	key, err := secp256k1.ParsePubKey(b)
 	return &PublicKeySECP256K1R{
 		pk:    key,
-		bytes: b,
+		bytes: key.SerializeCompressed(),
```

Comment:
Would it make more sense to report an error if `len(b) != SECP256K1RPKLen`? It seems weird to accept the various formats when everything else assumes the compressed format.

Reply:
The default behavior with the underlying `ParsePubKey` function accepts any SEC key format. I'll add a check prior to that function call to assert that the length is 33.

Agree that only the compressed format should be accepted in this call to avoid confusion.

Reply:
With the length check, do we still need to call `key.SerializeCompressed()`? As is, `err` may be non-nil, which would result in a nil panic.